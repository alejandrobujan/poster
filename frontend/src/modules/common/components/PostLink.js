import {Link} from 'react-router-dom';

const PostLink = ({id, title}) => {

    return (
        <Link to={`/post/post-details/${id}`}>
            {title}
        </Link>
    );

}

export default PostLink;