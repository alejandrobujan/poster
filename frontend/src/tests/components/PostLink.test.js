import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import PostLink from '../../modules/common/components/PostLink';
import '@testing-library/jest-dom/extend-expect';

describe('PostLink Component', () => {
  it('should render without crashing', () => {
    const { getByText } = render(
      <MemoryRouter>
        <PostLink id={1} title="Test Post" />
      </MemoryRouter>
    );
    expect(getByText('Test Post')).toBeInTheDocument();
  });

  it('should render a Link with the correct "to" prop', () => {
    const { getByText } = render(
      <MemoryRouter>
        <PostLink id={123} title="Test Post" />
      </MemoryRouter>
    );
    
    const link = getByText('Test Post').closest('a');
    expect(link).toHaveAttribute('href', '/post/post-details/123');
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <PostLink id={1} title="Test Post" />
      </MemoryRouter>
    );
    
    expect(asFragment()).toMatchSnapshot();
  });
});
