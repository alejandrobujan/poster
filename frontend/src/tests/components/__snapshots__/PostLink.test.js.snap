// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostLink Component should match the snapshot 1`] = `
<DocumentFragment>
  <a
    href="/post/post-details/1"
  >
    Test Post
  </a>
</DocumentFragment>
`;
